### PR TITLE
CLI-13882: Fix dialog InUse not being set to false when Dialog is not in use

### PR DESCRIPTION
**Pull request by DarraghGriffin**

### CODE REVIEW

File: CoreScriptsRoot/ServerStarterScript.lua [hunk 1]
```diff
@@ -24,13 +24,34 @@ RemoteEvent_SetDialogInUse.Name = "SetDialogInUse"
 RemoteEvent_SetDialogInUse.Parent = RobloxReplicatedStorage
 
 --[[ Event Connections ]]--
-local function setDialogInUse(player, dialog, value)
+local playerDialogMap = {}
+
+local function setDialogInUse(player, dialog, value, waitTime)
```
Comment (hockeychaos) [inline CoreScriptsRoot/ServerStarterScript.lua]: should we flag these changes?

File: CoreScriptsRoot/CoreScripts/MainBotChatScript2.lua [hunk 2]
```diff
@@ -43,9 +43,6 @@ function waitForChild(instance, name)
 	end
 end
 
-local filteringEnabledFixFlagSuccess, filteringEnabledFixFlagValue = pcall(function() return settings():GetFFlag("FilteringEnabledDialogFix") end)
```
Comment (DarraghGriffin) [inline CoreScriptsRoot/CoreScripts/MainBotChatScript2.lua]: Yes, this flag has been on for several months.

File: CoreScriptsRoot/CoreScripts/MainBotChatScript2.lua [hunk 3]
```diff
@@ -566,7 +537,7 @@ end
 
 function addDialog(dialog)
 	if dialog.Parent then
-		if dialog.Parent:IsA("BasePart") then
+		if dialog.Parent:IsA("BasePart") and dialog:IsDescendantOf(game.Workspace) then
```
Comment (DarraghGriffin) [inline CoreScriptsRoot/CoreScripts/MainBotChatScript2.lua]: We don't want the dialog GUI to display when the dialog object is parented to another service such as ReplicatedStorage or Lighting.

See report here: http://devforum.roblox.com/t/is-dialog-still-supported/29359/7

File: CoreScriptsRoot/ServerStarterScript.lua (hunk 1 above)
Comment (DarraghGriffin) [inline CoreScriptsRoot/ServerStarterScript.lua]: Added a flag

**Pull request merged**